src/test/java/org/broadinstitute/consent/http/resources/DacResourceTest.java

```diff
@@ -0,0 +1,197 @@
+package org.broadinstitute.consent.http.resources;
+
+import org.broadinstitute.consent.http.models.Dac;
+import org.broadinstitute.consent.http.models.DacBuilder;
+import org.broadinstitute.consent.http.service.DacService;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.List;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+public class DacResourceTest {
+
+    @Mock
+    DacService dacService;
+
+    private DacResource dacResource;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        dacResource = new DacResource(dacService);
+    }
+
+    @Test
+    public void testFindAll_success_1() {
+        when(dacService.findAll()).thenReturn(Collections.emptyList());
+
+        Response response = dacResource.findAll();
+        Assert.assertEquals(200, response.getStatus());
+        List dacs = ((List) response.getEntity());
+        Assert.assertTrue(dacs.isEmpty());
+    }
+
+    @Test
+    public void testFindAll_success_2() {
+        Dac dac = new DacBuilder()
+                .setName("name")
+                .setDescription("description")
+                .build();
+        when(dacService.findAll()).thenReturn(Collections.singletonList(dac));
+
+        Response response = dacResource.findAll();
+        Assert.assertEquals(200, response.getStatus());
+        List dacs = ((List) response.getEntity());
+        Assert.assertEquals(1, dacs.size());
+    }
+
+    @Test
+    public void testCreateDac_success() throws Exception {
+        Dac dac = new DacBuilder()
+                .setName("name")
+                .setDescription("description")
+                .build();
+        when(dacService.createDac(any(), any())).thenReturn(1);
+        when(dacService.findById(1)).thenReturn(dac);
+
+        Response response = dacResource.createDac(dac);
+        Assert.assertEquals(200, response.getStatus());
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void testCreateDac_badRequest_1() throws Exception {
+        Dac dac = null;
+        when(dacService.createDac(any(), any())).thenReturn(1);
+        when(dacService.findById(1)).thenReturn(dac);
```

Comment:
do you need these 2 lines of code? the createDac request doesn't get to either call. same with next couple of tests below

Reply:
Maybe not ... probably a copy/paste thing. Will fix, thank you!